json-endpoints/class.wpcom-json-api-get-comment-counts-endpoint.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/*
+ * WARNING: This file is distributed verbatim in Jetpack.
+ * There should be nothing WordPress.com specific in this file.
+ *
+ * @hide-in-jetpack
+ */
+
+new WPCOM_JSON_API_GET_Comment_Counts_Endpoint( array(
+	'description'   => 'Get comment counts for each available status',
+	'group'         => 'comments',
+	'stat'          => 'comments:1:comment-counts',
+	'method'        => 'GET',
+	'path'          => '/sites/%s/comment-counts',
+	'path_labels'   => array(
+		'$site'       => '(int|string) Site ID or domain',
+	),
+
+	'query_parameters' => array(
+		'post_id' => '(int) post ID for filtering the comment counts by post',
+	),
+
+	'example_request' => 'https://public-api.wordpress.com/rest/v1/sites/en.blog.wordpress.com/comment-counts',
+
+	'response_format' => array(
+		'counts' => array(
+			'approved' => '(int) Number of approved comments',
+			'awaiting_moderation' => '(int) Number of unapproved comments',
+			'trash' => '(int) Number of trash comments',
+			'spam' => '(int) Number of spam comments',
+			'post-trashed' => '(int) Number of comments whose parent post has been trashed',
+			'total_comments' => '(int) Combined number of comments in each category',
+			'all' => '(int) Combined number of approved and awaiting_moderation comments',
+		)
+	)
+) );
+
+class WPCOM_JSON_API_GET_Comment_Counts_Endpoint extends WPCOM_JSON_API_Endpoint {
+
+	// /sites/%s/comment-counts
+	public function callback( $path = '', $blog_id = 0 ) {
+		$blog_id = $this->api->switch_to_blog_and_validate_user( $this->api->get_blog_id( $blog_id ) );
+
+		if ( is_wp_error( $blog_id ) ) {
+			return $blog_id;
+		}
+
+		if ( ! get_current_user_id() ) {
+			return new WP_Error( 'authorization_required', 'An active access token must be used to retrieve comment counts.', 403 );
+		}
+
+		if ( ! current_user_can_for_blog( $blog_id, 'moderate_comments' ) ) {
```

Comment:
So moderate_comments is sadly not used anyplace but an obscure ajax endpoint for wp-admin. https://github.com/Automattic/wp-calypso/issues/17687

The corner case to check for is an Author who has a post with some comments. In wp-admin I should be able to see full counts, but only be able to moderate comments on their posts. In this case I think we need to switch this check to 'edit_posts'

Reply:
Updated in bb1606d219bfab8522dd9c9b888862c74d11478c.